core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -503,7 +503,9 @@ public synchronized void startInternal() throws InterruptedException, IOExceptio
   @Override
   public synchronized void stopInternal() throws InterruptedException, IOException {
     LOG.info("Shutting down raft journal");
-    mRaftJournalWriter.close();
+    if (mRaftJournalWriter != null) {
```

Comment:
Do you know in what scenario the writer would be `null` but `stopInternal ` is called?

Reply:
When the journal is closed while it's secondary.
Writer is initialized when it's primary.

Reply:
Would this happen when a secondary is promoted to a primary?

Reply:
During promotion writer is instantiated.